var/spack/repos/builtin/packages/py-pynn/package.py

```diff
@@ -15,20 +15,30 @@ class PyPynn(PythonPackage):
     pypi = "PyNN/PyNN-0.8.3.tar.gz"
     git = "https://github.com/NeuralEnsemble/PyNN.git"
 
+    version("0.10.0", sha256="04120fe0e03260d664b337e0ac29d985c3fb3684ef35b1add93a66739891c98f")
     version("0.9.1", sha256="bbc60fea3235427191feb2daa0e2fa07eb1c3946104c068ac8a2a0501263b0b1")
     version("0.8.3", sha256="9d59e6cffa4714f0c892ec6b32d1f5f8f75ba3a20d8635bac50c047aa6f2537e")
     version("0.8beta", commit="ffb0cb1661f2b0f2778db8f71865978fe7a7a6a4")
     version("0.8.1", sha256="ce94246284588414d1570c1d5d697805f781384e771816727c830b01ee30fe39")
     version("0.7.5", sha256="15f75f422f3b71c6129ecef23f29d8baeb3ed6502e7a321b8a2596c78ef7e03c")
 
     depends_on("python@2.6:2.8,3.3:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"), when="@0.10.0:")
+
     # pip silently replaces distutils with setuptools
     depends_on("py-setuptools", type="build")
     depends_on("py-jinja2@2.7:", type=("build", "run"))
     depends_on("py-docutils@0.10:", type=("build", "run"))
```

Comment:
jinja2 and docutils are no longer needed in the latest release

Reply:
sorry, for my understanding, spack itself handles those two dependencies or what do you mean?

Reply:
I mean py-pynn 0.9.1 depends on these, but py-pynn 0.10.0 does not. So you should add `when="@:0.9"` to them.

Reply:
they are at least still part of the requirements file https://github.com/NeuralEnsemble/PyNN/blob/0.10.0/requirements.txt
where do you get the information that they are not required anymore? I also don't see anything in the release notes

Reply:
The requirements file isn't used by pip/build/any package managers and is more or less meaningless. See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#dependencies for a list of locations to check for dependencies. In this case, `setup.py` is the file that lists the dependencies of the project, all other files are ignored.

Reply:
ok, thanks for the input and the review